src/xwayland/xwm/surface.rs

```diff
@@ -241,7 +248,7 @@ impl X11Surface {
     /// Returns if the window is currently mapped or not
     pub fn is_mapped(&self) -> bool {
         let state = self.state.lock().unwrap();
-        state.wl_surface.is_some() || state.mapped_onto.is_some()
+        state.mapped_onto.is_some()
```

Comment:
I am not sure this is right for override-redirect windows. We don't set the mapped_onto for those, but those can still be mapped/unmapped in which case Xwayland will remove/re-create the surface.

In general it is not a correct assumption, that an X-window will only have a single wl_surface associated during it's whole lifetime.

Reply:
I'm not sure about what the context for this function is, but it seems like it was wrong before as well?

Reply:
yeah I guess just checking for the surface to exist would be enough?

Reply:
Changed, but I'm not sure I understand. :smile:

Reply:
You did in fact understand me correctly. :D